src/metrics/runtime_metrics.rs

```diff
@@ -0,0 +1,148 @@
+use crate::metrics::meter::Meter;
```

Comment:
As mentioned in the larger comment. Get rid of MetricValue and individual counters and IncomingMessageRate struct and use ``metrics`` macros directly where possible or use ``Meter`` directly instead of wrapped in ``IncomingMessageRate``.